h5py/_hl/group.py

```diff
@@ -217,17 +231,21 @@ def create_dataset_like(self, name, other, **kwupdate):
         if other.maxshape != other.shape:
             kwupdate.setdefault('maxshape', other.maxshape)
 
-        return self.create_dataset(name, **kwupdate)
+        return self.create_dataset(name, dcpl=other._dcpl, **kwupdate)
+
+    def require_group(self, name, **kwargs):
```

Comment:
I am also 50/50 on this.  On one hand it brings it to parity with `require_dataset`, on the other hand, I have issues with `require_dataset` (https://github.com/h5py/h5py/pull/897). 

How critical is this to this PR?

Reply:
I can revert this and mark TODO: this part is not *absolutely* critical for the core feature.
Is it fine?

Reply:
I'd say that's OK.